src/libsyntax/parse/parser/item.rs

```diff
@@ -160,16 +160,27 @@ impl<'a> Parser<'a> {
         if self.eat_keyword(kw::Const) {
             let const_span = self.prev_span;
             if self.check_keyword(kw::Fn)
+                || self.check_keyword(kw::Extern)
                 || (self.check_keyword(kw::Unsafe)
-                    && self.is_keyword_ahead(1, &[kw::Fn])) {
+                    && (self.is_keyword_ahead(1, &[kw::Fn])
```

Comment:
Hmm... looking at this code again I believe the look-ahead can actually be removed because once we know that `const unsafe` has been parsed the continuation must eventually end up in `fn` for this to be a valid parse. With this said, I think we can rewrite this as `if [kw::Fn, kw::Unsafe, kw::Extern].iter().any(|k| self.check_keword(k)) {`.